### ISSUE TITLE
Document service typescript failure

### ISSUE THREAD

**Issue by guiguil81** [strapi/strapi#21879]
Hello ! 

I have a issue with typescript and my strapi document service.

- Node.js version: 20.16.0
- NPM version: 10.8.2
- Strapi version: 5.1.0
- Database: Sqlite
- Operating system: MacOS
- Is your project Javascript or Typescript: Typescript

I have this code

`
export default async function getCurrentRound() {
  return await strapi.documents('api::round.round').findMany({
    filters: {
      isActive: true,
    },
    fields: ['isActive'],
    populate: {
      games: {
        fields: ['planetName']
      }
    },
  })
}
`

And this

`
import getCurrentRound from "../getters/getCurrentRound";
import {ApiRoundRound} from "../../types/generated/contentTypes";

// const limit = 9007199254740991;

export default module.exports = {
  '*/10 * * * * *': async () => {
    console.time('ExecutionCron');

    const currentRound = await getCurrentRound();

    if (currentRound.length === 0) return null;

    currentRound.forEach(round => {
      round.games.forEach(game => {
        console.log(game)
      })
    })

    // console.log('currentRound', currentRound)
    setTimeout(() => {
      console.log('hehe');
    }, 2000)

    console.timeEnd('ExecutionCron');
  },
};
`

My const currentRound return me this type:

`
export default async function getCurrentRound(): Promise<{id: ID, documentId: string, locale?: string, name?: string, isActive?: boolean, endDate?: DateTimeValue, createdAt?: DateTimeValue, updatedAt?: DateTimeValue, publishedAt?: DateTimeValue}[]>
`

And round.games is not recognised

When i update fields: ['isActive'] that not update type and population is not recognised

**Comment by Convly** [comment]
I tried to reproduce the issue with your content-type, no components and the same function/queries, but couldn't force the error

The only difference is that I'm targeting `admin::api-token` instead of `api::game.game` for the `games` attribute since I didn't have access to its definition.

```typescript
export interface ApiRoundRound extends Struct.CollectionTypeSchema {
  collectionName: 'rounds';
  info: {
    singularName: 'round';
    pluralName: 'rounds';
    displayName: 'Round';
    description: '';
  };
  options: {
    draftAndPublish: false;
  };
  attributes: {
    name: Schema.Attribute.String & Schema.Attribute.Required & Schema.Attribute.Unique;
    isActive: Schema.Attribute.Boolean & Schema.Attribute.DefaultTo<false>;
    endDate: Schema.Attribute.DateTime & Schema.Attribute.Required;
    games: Schema.Attribute.Relation<'oneToMany', 'admin::api-token'>;
    createdAt: Schema.Attribute.DateTime;
    updatedAt: Schema.Attribute.DateTime;
    publishedAt: Schema.Attribute.DateTime;
    createdBy: Schema.Attribute.Relation<'oneToOne', 'admin::user'> & Schema.Attribute.Private;
    updatedBy: Schema.Attribute.Relation<'oneToOne', 'admin::user'> & Schema.Attribute.Private;
    locale: Schema.Attribute.String & Schema.Attribute.Private;
    localizations: Schema.Attribute.Relation<'oneToMany', 'api::round.round'> &
      Schema.Attribute.Private;
  };
}
```


![image](https://github.com/user-attachments/assets/57c4d2ec-5845-4e87-a5b7-adef94f28e18)
![image](https://github.com/user-attachments/assets/6a6fdf50-6e7b-44af-9c00-1de36d831335)

**Comment by Convly** [comment]
What happens if you manually change the relation target in the type definition?
Could you maybe share the definition of `ApiGameGame` too?

If none of that helps, maybe sharing a minimal working reproduction example (with everything sensitive removed) could allow us to reproduce and better understand the issue

**Comment by guiguil81** [comment]
I push a fake project with same database here tel me if is good for you

Also i use webstorem ide, maybe i can try with other ide

https://github.com/guiguil81/typeErrorStrapi

**Comment by Convly** [comment]
And same for the result of getCurrentRound
![image](https://github.com/user-attachments/assets/16fa92bf-8339-4ba2-80a1-a0d8b4b18af8)
![image](https://github.com/user-attachments/assets/767256e6-b7d8-4f7d-9cc7-d57de1083097)

**Comment by guiguil81** [comment]
I test on visual studio and i have this

```
const currentRound: {
    id: ID;
    documentId: DocumentID;
    isActive?: boolean;
    games?: {
        id: ID;
        documentId: DocumentID;
        round?: {
            id: ID;
            documentId: DocumentID;
            ... 10 more ...;
            localizations?: ...[];
        };
        ... 15 more ...;
        gameHaveSpecies?: {
            ...;
        }[];
    }[];
}[]
```

Sorry i thinks my webstorm is broken

**Comment by guiguil81** [comment]
That's is. normal to have all filed on game with this ?

```
export default async function getCurrentRound() {
  return await strapi.documents('api::round.round').findMany({
    filters: {
      isActive: true,
    },
    fields: ['isActive'],
    populate: {
      games: {
        fields: ['planetName'],
        populate: []
      }
    },
  })
}
```

In game i just have here planetName fields but type propose me all field, but that is a detail

**Comment by guiguil81** [comment]
En utilisant Yarn tout fonctionne comme dans tes screen en tout cas

**Comment by Convly** [comment]
> That's is. normal to have all filed on game with this ?
> 
> In game i just have here planetName fields but type propose me all field, but that is a detail

Yes, params on nested populate clauses are not taken into account by the types yet. (it returns a partial object with every property)

**Comment by Convly** [comment]
Closing this as the issue seems to be config-related.
Don't hesitate to re-open it if you encounter any further problems.

Have a nice day!
